habitat_baselines/il/trainers/pacman_trainer.py

```diff
@@ -0,0 +1,663 @@
+                            )
+
+                            metrics.dump_log()
+
+                    # Dataloader length for IterableDataset doesn't take into
+                    # account batch size for Pytorch v < 1.6.0
+                    num_batches = math.ceil(
+                        len(nav_dataset) / config.IL.NAV.batch_size
+                    )
+
+                    avg_p_loss /= len(num_batches)
+                    avg_c_loss /= len(num_batches)
+
+                    end_time = time.time()
+                    time_taken = "{:.1f}".format((end_time - start_time) / 60)
+
+                    logger.info(
+                        "Epoch {} completed. Time taken: {} minutes.".format(
+                            epoch, time_taken
+                        )
+                    )
+
+                    logger.info(
+                        "Average planner loss: {:.2f}".format(avg_p_loss)
+                    )
+                    logger.info(
+                        "Average controller loss: {:.2f}".format(avg_c_loss)
+                    )
+
+                    print("-----------------------------------------")
+
+                    if epoch % config.CHECKPOINT_INTERVAL == 0:
+                        self.save_checkpoint(
+                            model.state_dict(), "epoch_{}.ckpt".format(epoch)
+                        )
+
+                    epoch += 1
+
+    def _eval_checkpoint(
+        self,
+        checkpoint_path: str,
+        writer: TensorboardWriter,
+        checkpoint_index: int = 0,
+    ) -> None:
+        r"""Evaluates a single checkpoint.
+
+        Args:
+            checkpoint_path: path of checkpoint
+            writer: tensorboard writer object for logging to tensorboard
+            checkpoint_index: index of cur checkpoint for logging
+
+        Returns:
+            None
+        """
+        config = self.config
+
+        config.defrost()
+        config.TASK_CONFIG.DATASET.SPLIT = self.config.EVAL.SPLIT
+        config.freeze()
+
+        with habitat.Env(config.TASK_CONFIG) as env:
+            nav_dataset = NavDataset(
+                config,
+                env,
+                self.device,
+            ).decode("rgb")
+
+            nav_dataset = nav_dataset.map(nav_dataset.map_dataset_sample)
+
+            eval_loader = DataLoader(nav_dataset)
+
+            logger.info("eval_loader has {} samples".format(len(nav_dataset)))
+
+            q_vocab_dict, ans_vocab_dict = nav_dataset.get_vocab_dicts()
+
+            model_kwargs = {"q_vocab": q_vocab_dict.word2idx_dict}
+            model = NavPlannerControllerModel(**model_kwargs)
+
+            invalids = []
+
+            state_dict = torch.load(checkpoint_path)
+            model.load_state_dict(state_dict)
+            model.eval().to(self.device)
+
+            results_dir = config.RESULTS_DIR.format(split="val")
+
+            metrics = NavMetric(
+                info={"split": "val"},
+                metric_names=[
```

Comment:
This can probably be simplified with a few simple list comprehensions and format string